core/src/data_lookup/mod.rs

```diff
@@ -33,6 +33,7 @@ pub enum Error {
 )]
 pub struct DataLookup {
 	pub(crate) index: Vec<(AppId, DataLookupRange)>,
+	pub(crate) is_error: bool,
```

Comment:
What about making `DataLookup` optional in the header instead of adding `is_error`?

Reply:
Or even Result?

Reply:
I'm not in favour of doing drastic changes again, any thoughts @markopoloparadox ?

Reply:
As discussed in the previous call, we cannot touch `CompactDataLookup` for the time being. Making DataLookup a Option or Result will not yield the desired outcome since we need that struct to valid to decode it into  `CompactDataLookup`.

IE it easier to add a flag and create a corresponding  `CompactDataLookup` out of it then to write special functions to handle Option<DataLookup> into `CompactDataLookup` conversion and vice versa